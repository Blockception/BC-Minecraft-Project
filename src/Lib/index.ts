/*	Auto generated	*/

export * from "./mcattributes";
export * from "./mcdefinitions";
export * from "./mcignore";
export * from "./project";