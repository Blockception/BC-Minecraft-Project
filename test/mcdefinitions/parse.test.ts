import { TestFilesFolder } from "../utillity";
import * as path from "path";
import { MCDefinition } from "../../src/main";

const Text1 = `tag=allowed
tag=!denied`;

describe("MCDefinition", () => {
  it("parse1", () => {
    const definitions = MCDefinition.parse(Text1);
    expect(definitions).toMatchSnapshot();
  });

  it("loadSync file1", () => {
    const filepath = path.join(TestFilesFolder, "mcdefinitions", "file1.mcdefinitions");
    const definitions = MCDefinition.loadSync(filepath);
    expect(definitions).toMatchSnapshot();
  });

  it("load file2", async () => {
    const filepath = path.join(TestFilesFolder, "mcdefinitions", "file2.mcdefinitions");

    const definitions = await MCDefinition.load(filepath);
    expect(definitions).toMatchSnapshot();
  });
});
