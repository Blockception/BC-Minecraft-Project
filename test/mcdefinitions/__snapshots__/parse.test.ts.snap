// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`MCDefinition load file2 1`] = `
{
  "family": {
    "defined": [
      "npc",
      "Npc",
      "enemy",
    ],
    "excluded": [
      "nemesis",
    ],
  },
  "name": {
    "defined": [
      "Steve",
    ],
    "excluded": [
      "Creeper",
    ],
  },
  "objective": {
    "defined": [
      "counter",
    ],
    "excluded": [
      "Foo",
    ],
  },
  "tag": {
    "defined": [
      "tag_number1",
    ],
    "excluded": [
      "FoLolOl",
    ],
  },
}
`;

exports[`MCDefinition loadSync file1 1`] = `
{
  "family": {
    "defined": [
      "npc",
      "Npc",
      "enemy",
    ],
    "excluded": [
      "nemesis",
    ],
  },
  "name": {
    "defined": [
      "Steve",
    ],
    "excluded": [
      "Creeper",
    ],
  },
  "objective": {
    "defined": [
      "counter",
    ],
    "excluded": [
      "Foo",
    ],
  },
  "tag": {
    "defined": [
      "tag_number1",
    ],
    "excluded": [
      "FoLolOl",
    ],
  },
}
`;

exports[`MCDefinition parse1 1`] = `
{
  "tag": {
    "defined": [
      "allowed",
    ],
    "excluded": [
      "denied",
    ],
  },
}
`;
